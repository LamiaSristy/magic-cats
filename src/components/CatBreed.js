/*eslint-disable */
import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import '../stylesheets/catbreed.scss';

const Cat = ({ cat, history }) => {
  const showCat = id => history.push(`/cat/${id}`);
  return (
    <div className="cat-breed">
      <div className="cat-container"
          onClick={() => showCat(cat.id)}
          onKeyDown={() => showCat(cat.id)}
          role="button"
      >      
        <img src={cat.url} alt={cat.name} className="img" />  
        <p className="cat-name">{cat.name}</p>
      </div>
    </div>
  )
};

Cat.propTypes = {
  cat: {
    id: PropTypes.string,
    name: PropTypes.string,
    url: PropTypes.string,
  }.isRequired,
  history: PropTypes.instanceOf(Object).isRequired,
};

export default withRouter(Cat);