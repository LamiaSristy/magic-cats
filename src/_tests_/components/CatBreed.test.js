/*eslint-disable*/
import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom/extend-expect';
import CatBreed from '../../components/CatBreed';

const setup = () => shallow(<CatBreed />);

describe('CatBreed component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });

  it('Should render', () => {
    const element = component.find('.cat-name');
    expect(element.length).toEqual(0);
  });

  it('Should render', () => {
    const elementImg = component.find('.img');
    expect(elementImg.length).toBeLessThan(10);
  });

  it('Should render', () => {
    const elementName = component.find('.cat-name');
    expect(elementName.length).toBeLessThan(5);
  });

  it('Should render', () => {
    const elementButton = component.find('.cat-breed');
    expect(elementButton).toBeVisible;
  });

  test('renders the cat-breed the page', () => {
    const catbreed = component.find('.cat-breed');
    expect(catbreed).toMatchSnapshot();
  });
  test('renders the cat-container the page', () => {
    const catcontainer = component.find('.cat-container');
    expect(catcontainer).toMatchSnapshot();
  });
  test('renders the cat-images the page', () => {
    const catimg = component.find('.img');
    expect(catimg).toMatchSnapshot();
  });
});
