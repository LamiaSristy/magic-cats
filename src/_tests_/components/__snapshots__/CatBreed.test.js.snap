// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatBreed component renders the cat-breed the page 1`] = `ShallowWrapper {}`;

exports[`CatBreed component renders the cat-breed the page 2`] = `ShallowWrapper {}`;

exports[`CatBreed component renders the cat-container the page 1`] = `ShallowWrapper {}`;

exports[`CatBreed component renders the cat-images the page 1`] = `ShallowWrapper {}`;

exports[`CatBreed component renders the pagenotfound the page 1`] = `ShallowWrapper {}`;
